tests/e2e/test_slice.py

```diff
@@ -34,9 +47,130 @@ def getslice(lineno, direction):
     return meta, data
 
 
-def test_inline_slice():
-    _, slice = getslice(5, 'inline')
-    print (slice)
-    assert len(slice) == 2
-    np.testing.assert_array_equal(slice[0], np.array([116, 117, 118, 119]))
-    np.testing.assert_array_equal(slice[1], np.array([120, 121, 122, 123]))
+@pytest.mark.parametrize("method", [
+    ("get"),
+    ("post")
+])
+def test_inline_slice(method):
+    meta, slice = request_slice(method, 5, 'inline')
+    expected = np.array([[116, 117, 118, 119],
+                         [120, 121, 122, 123]])
+    assert np.array_equal(slice, expected)
+
+    # what is format?
+    expected_meta = json.loads("""
+    {
+        "axis": [
+            {"Annotation": "Crossline", "max": 11.0, "min": 10.0, "samples" : 2, "unit": "unitless"},
+            {"Annotation": "Sample", "max": 16.0, "min": 4.0, "samples" : 4, "unit": "ms"}
+        ],
+        "format": 3
+    }
+    """)
+    assert meta == expected_meta
+
+
+@pytest.mark.parametrize("method", [
+    ("get"),
+    ("post")
+])
+def test_crossline_slice(method):
```

Comment:
I am not sure (hence the comment at the end of the file).
1. From practical point of view, these tests (and actually slice tests for crossline/sample) are not needed. They all are expected to be tested on the unit level. These e2e actually need to make sure that yet untested code works correctly - which is go bindings. And from looking-at-the code we can see that paths don't actually differ in case of slice/inline/sample. It's just a parameter. So it doesn't have to be tested here.
2. From theoretical point of view, you want to test from e2e everything important that user might encounter. Because what if there are changes in the unit code...
So that's why I added inline/xline/samples slice tests - as these endpoints are kinda important.

So I am kinda unsure what should be double-tested here and for what unit tests are totally enough.